vespalib/src/tests/net/tls/policy_checking_certificate_verifier/policy_checking_certificate_verifier_test.cpp

```diff
@@ -0,0 +1,187 @@
+    auto glob = HostGlobPattern::create_from_glob(pattern);
+    return glob->matches(string_to_check);
+}
+
+TEST("glob without wildcards matches entire string") {
+    EXPECT_TRUE(glob_matches("foo", "foo"));
+    EXPECT_FALSE(glob_matches("foo", "fooo"));
+    EXPECT_FALSE(glob_matches("foo", "ffoo"));
+}
+
+TEST("wildcard glob can match prefix") {
+    EXPECT_TRUE(glob_matches("foo*", "foo"));
+    EXPECT_TRUE(glob_matches("foo*", "foobar"));
+    EXPECT_FALSE(glob_matches("foo*", "ffoo"));
+}
+
+TEST("wildcard glob can match suffix") {
+    EXPECT_TRUE(glob_matches("*foo", "foo"));
+    EXPECT_TRUE(glob_matches("*foo", "ffoo"));
+    EXPECT_FALSE(glob_matches("*foo", "fooo"));
+}
+
+TEST("wildcard glob can match substring") {
+    EXPECT_TRUE(glob_matches("f*o", "fo"));
+    EXPECT_TRUE(glob_matches("f*o", "foo"));
+    EXPECT_TRUE(glob_matches("f*o", "ffoo"));
+    EXPECT_FALSE(glob_matches("f*o", "boo"));
+}
+
+TEST("wildcard glob does not cross multiple dot delimiter boundaries") {
+    EXPECT_TRUE(glob_matches("*.bar.baz", "foo.bar.baz"));
+    EXPECT_TRUE(glob_matches("*.bar.baz", ".bar.baz"));
+    EXPECT_FALSE(glob_matches("*.bar.baz", "zoid.foo.bar.baz"));
+    EXPECT_TRUE(glob_matches("foo.*.baz", "foo.bar.baz"));
+    EXPECT_FALSE(glob_matches("foo.*.baz", "foo.bar.zoid.baz"));
+}
+
+TEST("single char glob matches non dot characters") {
+    EXPECT_TRUE(glob_matches("f?o", "foo"));
+    EXPECT_FALSE(glob_matches("f?o", "fooo"));
+    EXPECT_FALSE(glob_matches("f?o", "ffoo"));
+    EXPECT_FALSE(glob_matches("f?o", "f.o"));
+}
+
+TEST("special basic regex characters are escaped") {
+    EXPECT_TRUE(glob_matches("$[.\\^", "$[.\\^"));
+}
+
+TEST("special extended regex characters are ignored") {
+    EXPECT_TRUE(glob_matches("{)(+|]}", "{)(+|]}"));
+}
+
+// TODO CN + SANs
+PeerCredentials creds_with_dns_sans(std::vector<vespalib::string> dns_sans) {
+    PeerCredentials creds;
+    creds.dns_sans = std::move(dns_sans);
+    return creds;
+}
+
+PeerCredentials creds_with_cn(vespalib::stringref cn) {
+    PeerCredentials creds;
+    creds.common_name = cn;
+    return creds;
+}
+
+bool verify(AllowedPeers allowed_peers, const PeerCredentials& peer_creds) {
+    auto verifier = create_verify_callback_from(std::move(allowed_peers));
+    return verifier->verify(peer_creds);
+}
+
+TEST("Default-constructed AllowedPeers does not allow all authenticated peers") {
+    EXPECT_FALSE(AllowedPeers().allows_all_authenticated());
+}
+
+TEST("Specially constructed set of policies allows all authenticated peers") {
+    auto allow_all = AllowedPeers::allow_all_authenticated();
+    EXPECT_TRUE(allow_all.allows_all_authenticated());
+    EXPECT_TRUE(verify(allow_all, creds_with_dns_sans({{"anything.goes"}})));
+}
+
+TEST("Non-empty policies do not allow all unauthenticated peers") {
+    auto allow_not_all = allowed_peers({policy_with({required_san_dns("hello.world")})});
+    EXPECT_FALSE(allow_not_all.allows_all_authenticated());
```

Comment:
the test says both authenticated and unauthenticated...

Reply:
👀 will fix, should be "authenticated" for both